docs/usage/schema.md

```diff
@@ -71,6 +71,8 @@ It has the following arguments:
   JSON Schema
 * `max_length`: for string values, this adds a corresponding validation and an annotation of `maxLength` to the
   JSON Schema
+* `read_only`: a boolean which defaults to False. When True, the field raises a TypeError if the field is 
```

Comment:
The name is misleading, (which I think is why the schema problem has come up). 

The function is not "read only" since the field can be set when creating a model. The name should be `allow_mutation` with default `True`. This is both a better description, and matches `Config.allow_mutation`.

Reply:
I agree allow_mutation is a better fit and less confusing. Will commit the changes shortly.

Reply:
@samuelcolvin - how do you feel about the TypeError being raised?  I'm open to other suggestions if you have one.